blobstore.wit.md

```diff
@@ -0,0 +1,186 @@
+```
+
+// # wasi-cloud blobstore service definition
+//
+// A blobstore is service that can mange containers and objects.
+//
+// ## Object
+// An object is a named sequence of bytes. Objects can be read or written with a stream interface.
+// Once an object is written its contents cannot be modified.
+// Objects can be any size from 1 byte up to the limits imposed by the underlying store.
+// Object names are unique within their container.
+//
+// ## Container
+// A container is a named collection of objects. Container names can be any utf-8 string.
+// Within any instance of the blobstore interface, all containers share the same "flat" namespace.
+
+
+// wasi-cloud Blobstore service definition
+interface "wasi:blob/blobstore" {
+
+  use wasi:blob/types::{ Error, container-metadata, container-name, object-id }
+
+  // creates a new empty container
+  create-container: func(name: container-name) -> result<container,Error>
+
+  // get a list of all containers, as a stream.
+  list-containers: func() -> result<stream<container-name>, Error>
```

Comment:
This I think borders on the edge here. A good deal of blob store implementations deliberately do not allow enumeration of buckets/containers.

Reply:
removed.